core/metrics_manager.js

```diff
@@ -0,0 +1,385 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Calculates and reports workspace metrics.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+'use strict';
+
+goog.provide('Blockly.MetricsManager');
+
+goog.require('Blockly.utils.Size');
+
+goog.requireType('Blockly.utils.Metrics');
+goog.requireType('Blockly.utils.toolbox');
+
+
+/**
+ * The manager for all workspace metrics calculations.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace to calculate metrics
+ *     for.
+ * @constructor
+ */
+Blockly.MetricsManager = function(workspace) {
+  /**
+   * The workspace to calculate metrics for.
+   * @type {!Blockly.WorkspaceSvg}
+   * @protected
+   */
+  this.workspace_ = workspace;
+};
+
+/**
+ * Describes the width, height and location of the toolbox on the main
+ * workspace.
+ * @typedef {{
+ *            width:number,
```

Comment:
We aren't consistent in our code base, but I think there should be a space after the `:` in these typedefs. So, for example, this would change to `width: number`.